core/src/avm2/globals/flash/display/shader_job.rs

```diff
@@ -134,27 +136,65 @@ pub fn get_shader_args<'gc>(
                                 bitmap.bitmap_handle(activation.gc(), activation.context.renderer),
                             )
                         } else if let Some(byte_array) = input.as_bytearray() {
-                            let expected_len = (width * height * input_channels) as usize
-                                * std::mem::size_of::<f32>();
-                            assert_eq!(byte_array.len(), expected_len);
                             assert_eq!(byte_array.endian(), Endian::Little);
-                            ImageInputTexture::Bytes {
+
+                            let num_pixels = (width * height) as usize;
+
+                            let (bytes, _trailing) = byte_array.bytes().as_chunks::<4>();
```

Comment:
You can use `_` instead of `_trailing` to let the compiler know the value can be dropped right away.